index.html

```diff
@@ -10711,7 +10711,47 @@ <h2>Definitions of States and Properties (all aria-* attributes)</h2>
 				</tbody>
 			</table>
 		</div>
-                <div class="property" id="aria-details">
+		<div class="property" id="aria-description">
+			<pdef>aria-description</pdef>
+			<div class="property-description">
+				<p>Defines a string value that describes or annotates the current element. See related <pref>aria-describedby</pref>.</p>
+				<p>The <pref>aria-description</pref> attribute is similar to the <pref>aria-label</pref> in that both provide a flat string to associate with the element, but a label should be concise, where a description is intended to provide more verbose information.</p>
+				<p>The purpose of <pref>aria-description</pref> is the same as that of <pref>aria-describedby</pref>. It provides the user with additional descriptive text for the object. The most common <a>accessibility <abbr title="Application Programing Interfaces">API</abbr></a> mapping for a description is the <a>accessible description</a> property.</p>
+				<p>If the description text is available in the content, authors SHOULD NOT use <pref>aria-description</pref>, but should use one of the following instead:</p>
+				<ul>
+					<li>Authors SHOULD use <pref>aria-describedby</pref> when the related description or annotation elements are hidden, or contain a simple, small description that is best experienced as a flat string, rather than by having the user navigate to them.</li>
```

Comment:
TL;DR: I think we want to remove "are hidden, or". This is my conclusion after typing out the following:

I find this somewhat confusing for a few reasons:

1. "available in the content" (in the paragraph above) tends to suggest "rendered" -- at least to me. In contrast "when the related description or annotation elements are hidden" tends to suggest not rendered and (in my mind) NOT "available in the content." Perhaps we can tweak the language a tad?

2. assuming "are hidden" means "not rendered", isn't there an excellent chance the element containing the description will be pruned from the accessibility tree? In which case, at least some platform accessibility APIs will wind up with an accessible described-by relationship pointing to a null target. Relationship targets should ideally not be null.

    Granted, this winds up being not a problem for most platforms because user agents are expected to flatten the element referenced by `aria-describedby` and expose it via the platform's accessible description property. As long as ATs prefer that property, there's no need to use the platform relationship. So I guess my main objection here is one of aesthetics. :grinning:

    That said, why would an author want to create hidden / not-rendered elements and then reference them via an ARIA property for the purpose of causing user agents to piece the description together as a flat string? *IF* the description is hidden / not-rendered, wouldn't it be a lot simpler for all parties to just use the new `aria-description`?

3. assuming "are hidden" means "not rendered", then doesn't the text here contradict what we state for `aria-label`/`aria-labelledby`? The spec text states:

    > If the interface is such that it is not possible to have a visible label on the screen, authors SHOULD use aria-label and SHOULD NOT use aria-labelledby.

Taking the above three observations into account, I would think that authors SHOULD use the new `aria-description` property and NOT `aria-describedby` "when the related description or annotation elements are hidden."

Or am I missing something?

Reply:
I still think this is good text, perhaps instead of "available in the content", we say "available in the DOM".
However, I think the first option of these 2 is slightly preferable:
1.
```
<p id="desc">The quick brown fox jumps over the lazy dog abcdef ..... 100 more words ..."</p>
<input type="text" aria-describedby="desc">
```
2.
```
<p>The quick brown fox jumps over the lazy dog abcdef ..... 100 more words ..."</p>
<input type="text" aria-description=">The quick brown fox jumps over the lazy dog abcdef ..... 100 more words ..."">
```

Benefits of version 2:
- Shorter markup
- No redundant text that could accidentally diverge
- In the hidden case, some user agents will expose a relationship, some will expose no relationship (as opposed to a null one, I haven't seen that happen). Note that Microsoft has been working on bringing hidden nodes into the exposed hierarchy for Chrome.

Interesting point about sounding like it conflicts with the aria-label/aria-labelledby text. The intention doesn't contradict. It isn't saying to put a hidden description in the DOM instead of using aria-description. It's saying that if a description is in the DOM, may as well use it.

Or?

Reply:
I opted to remove "are hidden, or", rather than copy the text from aria-label. The reason is that in this case there are 3 properties to chose from, and I like the way the bulleted list works.